### ISSUE TITLE
pkgin segfault if more than one repo is used

### ISSUE THREAD

**Issue by bahamat** [TritonDataCenter/pkgsrc-legacy#183]
I have a private repo that I use to distribute packages. Some are updates of packages in pkgsrc, some are new. The additional repo is appended to `/opt/local/etc/pkgin/repositories.conf`.

When I run `pkgin ug`, I get a segfault.

```
[root@node-c0546da1e5 ~]# tail -2 /opt/local/etc/pkgin/repositories.conf
http://pkgsrc.joyent.com/packages/SmartOS/2013Q4/x86_64/All
http://pkgsrc.example.com/packages/
[root@node-c0546da1e5 ~]# pkgin -f up
pkg_summary.bz2                                    100% 1741KB   1.7MB/s   1.7MB/s   00:00
cleaning database from http://pkgsrc.example.com/packages/ entries...
cleaning database from http://pkgsrc.joyent.com/packages/SmartOS/2013Q4/x86_64/All entries...
processing remote summary (http://pkgsrc.joyent.com/packages/SmartOS/2013Q4/x86_64/All)...
updating database: 100%
pkg_summary.gz                                     100%  703     0.7KB/s   0.7KB/s   00:00
processing remote summary (http://pkgsrc.example.com/packages/)...
updating database: 100%
[root@node-c0546da1e5 ~]# pstack /var/cores/core.pkgin.28432 
core '/var/cores/core.pkgin.28432' of 28432:    pkgin ug
 fffffd7fff186ed6 strcmp () + 16
 00000000004103a3 pkgin_upgrade () + 43
 000000000041f55d main () + 54d
 000000000040bc8c _start () + 6c
```

**Comment by bahamat** [comment]
Full truss output https://gist.github.com/bahamat/9720056

**Comment by bahamat** [comment]
This is also NetBSDfr/pkgin#1 and fixed in NetBSDfr/pkgin@338d0dd698da9a370f8bcffc1a6dda39b8147cac

**Comment by bahamat** [comment]
This is fixed in the 14.4.0 images.
